xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +80,36 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights and updates MSet.
+     *  Dereferencing the Iterator should return a double.
+     *  Iterator should dereference weights in the original order of documents.
+     *  Weights are assigned as per the original order of documents.
+     *
+     *  @param begin	Begin iterator.
+     *  @param end	End iterator.
+     *
+     *  @exception Xapian::InvalidArgument is thrown if the total number of
+     *  elements in the input dont match the total number of documents in MSet.
+     *
+     */
+    template <typename Iterator>
+    void replace_weights(Iterator first, Iterator last)
+    {
+	if (last - first != size()) {
+	    throw Xapian::InvalidArgumentError("Number of weights assigned doesn't match the number of items.");
+	}
+	Xapian::doccount i = 0;
+	while (first != last) {
+	    set_item_weight(i, *first);
+	    ++i, ++first;
+	}
+    }
+
+    /**
+     * Sorts the list of documents in MSet according to their weights. Use after calling MSet::replace_weights.
+     */
```

Comment:
This doc comment is poorly formatted.

The short description (i.e. the first sentence here) should be on the first line, then there should be a paragraph break between that and the extended description (which would avoid the overly long line).

The naming of `MSet::sort_by_relevance()` seems a bit too similar to `Enquire::set_sort_by_relevance()` - I'd be happy if it was clearer what this method was for from the name.  The obvious alternative is `resort_by_relevance()`, but "resort" is easy to misread as "resort" rather than "re-sort", and `re_sort_by_relevance()` looks weird.  Perhaps `rerank_by_relevance()` is better?  Thoughts?